src/fpmath.h

```diff
@@ -41,30 +41,23 @@
 #include "powerpc_fpmath.h"
 #endif
 
-#if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__)
-
 /* Definitions provided directly by GCC and Clang. */
-#define _LITTLE_ENDIAN    __ORDER_LITTLE_ENDIAN__
-#define _BIG_ENDIAN       __ORDER_BIG_ENDIAN__
-#define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
-#define _BYTE_ORDER       __BYTE_ORDER__
+#if !(defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && defined(__ORDER_BIG_ENDIAN__))
 
-#elif defined(__GLIBC__)
+#if defined(__GLIBC__)
```

Comment:
Given that I have no way to check other compilers, I'd rather keep these as it's not much code and it's pretty clear. Do you have any reference about what other compilers support? The only thing I have found is https://sourceforge.net/p/predef/wiki/Endianness/, which doesn't seem to say so.